lib/extensions/acts_as_helpers/active_record/base.rb

```diff
@@ -7,10 +7,22 @@ def acts_as_experience_points_record
       include ExperiencePointsInstanceMethods
     end
 
-    # Decorator for abstractions with concrete occurrences which have
-    # the potential to award course_users EXP Points
-    def acts_as_lesson_plan_item
+    # Decorator for abstractions with concrete occurrences for student activities with a start
+    #   and end date, with the option to: (i) Award course_users with EXP Points, and/or
+    #   (ii) Appear as a todo item for each course_user once it is published
+    #
+    # Todo acts as reminders for sutdents to act on. Todos are also automatically created
+    #   when the lesson_plan_item transits from draft to non-draft, and destroyed in the
+    #   reverse case (see Course::LessonPlan::ItemTodoConcern).
+    #
+    # To declare that an actable model has todos:
+    #   - Define has_todo to true when calling acts_as_lesson_plan_item
+    #   - Define hooks to update todo's workflow_state (see Course::LessonPlan::Todo)
+    def acts_as_lesson_plan_item(has_todo: false)
       acts_as :lesson_plan_item, class_name: Course::LessonPlan::Item.name
+
+      class_variable_set(:@@has_todo, has_todo ? true : false)
```

Comment:
Better not use class variables, it's considered evil and has weird behaviour in the inheritance case, use `attr_accessor` in its singleton class instead.